### ISSUE TITLE
no-native-reassign gives false errors

### ISSUE THREAD

**Issue by ljharb** [eslint/eslint#3153]
```js
function foo() {
  var valueOf = 3;
  return valueOf;
}
```

Assuming the function style is valid, `no-native-reassign` will error out here with `Redefinition of 'valueOf'       no-native-reassign`

However, because of `var valueOf`, the native isn't being reassigned. `valueOf` may be a global in some engines (and === to `Object.prototype.valueOf`) but unless I omit the `var`, I'm certainly not reassigning the native value - I'm simply shadowing it.

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by mysticatea** [comment]
Thank you for the report.

I could reproduce that.

Actually `no-native-reassign` is checking redeclarations and shadowing. So, we are going to move those checks to `no-redeclare` and `no-shadow` currently.

Please see #3070 and #3086.

**Comment by nzakas** [comment]
@mysticatea is this fixed by those two issues?

**Comment by mysticatea** [comment]
@nzakas Yes, I removed a check of shadowing from `no-native-reassign` in #3086.

I double checked:

- `no-native-reassign: 2` does not warn the code.
- `no-shadow: 2` does not warn the code.
- `no-shadow: [2, {builtinGlobals: true}]` warns the code. Because there is a variable `valueOf` in global (https://github.com/sindresorhus/globals/blob/master/globals.json#L58).

**Comment by nzakas** [comment]
Great, thanks!

**Comment by ljharb** [comment]
Cool, so this will be released in rc 4 then?

**Comment by mysticatea** [comment]
This is included in `v1.0.0-rc-3` :)

**Comment by ljharb** [comment]
oh huh, i was pretty sure the issue i was reporting was in rc-3. it may have been rc-2 tho, i'll confirm.
